### DISCUSSION TITLE
DropdownMenu with Content - closing on <Select>

### DISCUSSION THREAD

**Discussion by jzbuchalski** [iTwin/iTwinUI-react#998] [General]
Is it possible to prevent the dropdown (in a DropdownMenu) from closing when an option is selected in a Select?

![image](https://user-images.githubusercontent.com/17307464/208713375-8dd57d49-002f-4364-96e9-26a9e411528b.png)

**Comment by mayank99** [comment]
~~You can use `popoverProps.visible` to pass your own controlled state.~~

Edit: I misunderstood the question. So to clarify, the `DropdownMenu` is used with `MenuExtraContent` which contains a `Select`, and clicking on an option in the `Select` closes the `DropdownMenu`.

I think this is because of "close on outside click" behavior (the select menu is technically "outside" the `DropdownMenu` in the DOM). To fix this, it can be moved *inside* the `DropdownMenu` by passing `popoverProps={{ appendTo: 'parent' }}` (to `Select`).

This workaround is also briefly mentioned in the [FAQ](https://github.com/iTwin/iTwinUI-react/wiki/FAQ#popover-tooltips-dropdowns-etc) under a different problem.
![](https://user-images.githubusercontent.com/9084735/208750084-6490e9a9-3123-413a-8e64-50d3b9398501.png)

**Comment by jzbuchalski** [comment]
Ugh, was looking at the Select source, not the Dropdown.  Thanks.
